internal/onepassword/sdk/client.go

```
@@ -0,0 +1,180 @@
+	params := item.FromModelItemToSDKCreateParams()
+
+	sdkItem, err := c.sdkClient.Items().Create(ctx, params)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create item: %w", err)
+	}
+
+	modelItem := &model.Item{}
+	modelItem.FromSDKItemToModel(&sdkItem)
+	return modelItem, nil
+}
+
+func (c *Client) UpdateItem(ctx context.Context, item *model.Item, vaultUuid string) (*model.Item, error) {
+	currentItem, err := c.sdkClient.Items().Get(ctx, vaultUuid, item.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	params := item.FromModelItemToSDKCreateParams()
+	currentItem.Title = params.Title
+	currentItem.Category = params.Category
+	currentItem.Fields = params.Fields
+	currentItem.Sections = params.Sections
+	currentItem.Tags = params.Tags
+	currentItem.Websites = params.Websites
+	if params.Notes != nil {
+		currentItem.Notes = *params.Notes
+	}
+
+	updatedItem, err := c.sdkClient.Items().Put(ctx, currentItem)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert back to provider model
+	modelItem := &model.Item{}
+	modelItem.FromSDKItemToModel(&updatedItem)
+	return modelItem, nil
+}
+
+func (c *Client) DeleteItem(ctx context.Context, item *model.Item, vaultUuid string) error {
+	err := c.sdkClient.Items().Delete(ctx, vaultUuid, item.ID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) GetFileContent(ctx context.Context, file *model.ItemFile, itemUUID, vaultUUID string) ([]byte, error) {
+	fileAttributes := sdk.FileAttributes{
+		Name: file.Name,
+		ID:   file.ID,
+		Size: uint32(file.Size),
+	}
+
+	content, err := c.sdkClient.Items().Files().Read(ctx, vaultUUID, itemUUID, fileAttributes)
+	if err != nil {
+		return nil, err
+	}
+
+	return content, nil
+}
+
+func NewClient(ctx context.Context, config SDKConfig) (*Client, error) {
+	var sdkClient *sdk.Client
+	var err error
+
+	integrationName, integrationVersion, _ := strings.Cut(config.ProviderUserAgent, "/")
```

Let's check 3rd param, and if it's `false` let's return an error.

This way we are safe if somebody accidentally change user agent string.